core/domain/state_domain.py

```diff
@@ -1568,15 +1568,28 @@ def is_rte_content_supported_on_android(self):
         Returns:
             bool. Whether the RTE components in the state is valid.
         """
-        regex_string = r'oppia-noninteractive-'
-        regex_string += (
-            '|'.join(android_validation_constants.INVALID_RTE_COMPONENTS))
-        regex_string = regex_string[:-1]
-        regex = re.compile(regex_string)
-        if self.content and regex.search(self.content.html):
+        def whitelist_checker(html):
+            """Checks if the provided html string contains only whitelisted
+            RTE tags.
+
+            Args:
+                html: str. The html string.
+
+            Returns:
+                bool. Whether all RTE tags in the html are whitelisted.
+            """
+            rte_prefix = 'oppia-noninteractive-'
+            components = set([
+                rte['id'].replace(rte_prefix, '') for rte in html_cleaner
```

Comment:
Move html_cleaner to the start of the next line.

Reply:
Done.